mars/services/scheduling/worker/execution.py

```diff
@@ -48,16 +51,73 @@ class SubtaskExecutionInfo:
     supervisor_address: str
     result: SubtaskResult = field(default_factory=SubtaskResult)
     cancelling: bool = False
+    max_retries: int = 0
+    num_retries: int = 0
     slot_id: Optional[int] = None
     kill_timeout: Optional[int] = None
 
 
+class SlotContext:
```

Comment:
Now that we have stateless actors, context managers may be avoided.

Reply:
The stateless actor PR https://github.com/mars-project/mars/pull/2218 added a new API `wait_actor_pool_recovered` which is useful to wait the actor pool. But, I am not sure if it is related to the slot context.